### PR TITLE
feat: trigger that updates partition table when a parquet file is created

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Another PR for https://github.com/influxdata/idpe/issues/16836

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: iox_catalog/migrations/20230105120822_trigger_update_partition.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+END;
+$$;
+
+-- TRIGGER that fires the update_partition_on_new_file_at function when a new file is added to the parquet_file table
+CREATE TRIGGER update_partition 
+    AFTER INSERT ON parquet_file 
+    FOR EACH ROW 
+    EXECUTE PROCEDURE update_partition_on_new_file_at();
+    
+
```
Comment (NGA-TRAN) [inline iox_catalog/migrations/20230105120822_trigger_update_partition.sql:22]: I have tested this locally. When I insert a new parquet file, the correspondent partition is updated

File: iox_catalog/migrations/20230105120822_trigger_update_partition.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+-- A new field in the partition that stores the latest time a new file is added to the partition
+ALTER TABLE partition ADD COLUMN IF NOT EXISTS new_file_at bigint;
+
+-- FUNTION that updates the new_file_at field in the partition table when the update_partition trigger is fired
+CREATE OR REPLACE FUNCTION update_partition_on_new_file_at()
+RETURNS TRIGGER 
+LANGUAGE PLPGSQL
+AS $$
+BEGIN 
+    UPDATE partition SET new_file_at = (SELECT (EXTRACT(EPOCH FROM (select now() ))) * 1000000000) WHERE id = NEW.partition_id;
```
Comment (pauldix) [inline iox_catalog/migrations/20230105120822_trigger_update_partition.sql:10]: I think the selects here are extraneous. You should be able to just do: 
```
UPDATE partition SET new_file_at = extract (epoch from now()) * 1000000000 WHERE id = NEW.partition_id;
```

Comment (NGA-TRAN) [reply iox_catalog/migrations/20230105120822_trigger_update_partition.sql:10]: Good idea

Comment (NGA-TRAN) [reply iox_catalog/migrations/20230105120822_trigger_update_partition.sql]: Fixed @pauldix

**Review by pauldix — APPROVED** [review]

**Pull request merged**